app/models/race.rb

```diff
@@ -32,7 +32,7 @@ def name_with_event
   end
 
   def generate_slug
-    "#{starts_at.year} #{event.name} #{name}".parameterize
+    "#{starts_at.year} #{event.name} #{name}".parameterize if @race
```

Comment:
`@race` will always be `nil` here... not sure how this would ever have worked?

Reply:
I get this error without it:
![image](https://user-images.githubusercontent.com/16193879/100019134-2cd4e100-2da3-11eb-8218-d52180f8f625.png)

Reply:
Sounds like something that needs to be resolved with setting data in the factory. Really don't want to be using @vars in models.

Reply:
Thanks! I felt like that was a little bit hacky but I figured out a much better solution. I added a safe navigator operator to the generate_slug method and it worked like a charm.